GitCommands/Git/Executable.cs

```diff
@@ -111,9 +111,16 @@ private void OnProcessExit(object sender, EventArgs eventArgs)
                     // if the Process is disposed then reading ExitCode will throw.
                     if (!_disposed)
                     {
-                        var exitCode = _process.ExitCode;
-                        _logOperation.LogProcessEnd(exitCode);
-                        _exitTaskCompletionSource.TrySetResult(exitCode);
+                        try
+                        {
+                            var exitCode = _process.ExitCode;
+                            _logOperation.LogProcessEnd(exitCode);
+                            _exitTaskCompletionSource.TrySetResult(exitCode);
+                        }
+                        catch (Exception ex)
+                        {
+                            _logOperation.LogProcessEnd(ex);
```

Comment:
Should probably add this here so that any awaiters aren't blocked indefinitely.

```c#
_exitTaskCompletionSource.TrySetException(ex);
```

Reply:
Done.